go-get-tests.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+set -e
+
+echo "Running go get tests"
+
+# Keep existing state
+export ORIG_PATH=$PATH
+export ORIG_GIT=$(which git)
+
+export TEST_PKG=github.com/golang/protobuf/proto
+export TEST_GIT_DIR=../../golang/protobuf
+
+echo "Adding dgit to the path"
+go build
+mkdir bin
+cp dgit bin/git
+export PATH=$(pwd)/bin:$PATH
+
+export DGIT_TRACE=/tmp/go-get-dgit-log.$$.txt
+
+echo "Go get a package"
+go get ${TEST_PKG} || echo "Known problem with go get, it invokes the submodule command"
+test -d ${TEST_GIT_DIR} || (echo "ERROR: Go get didn't work"; exit 1)
+
+test -f $DGIT_TRACE || (echo "ERROR: Dgit wasn't called for the go get test"; exit 1)
+unset DGIT_TRACE
+
+echo "Reset the package back one commit from master"
+$ORIG_GIT -C ${TEST_GIT_DIR} reset HEAD^1
+$ORIG_GIT -C ${TEST_GIT_DIR} checkout .
```

Comment:
I don't get it. What is this for?

Reply:
It's to force it back one commit from origin/master so that pull can bring it back up to date. I used the official git to do this because I don't think we have the rev-parsing capability to handle the HEAD^1 (HEAD, backward one commit).

Reply:
I see. Makes sense. I'll merge this, and if there's any problems caused we can always revert it.